File: tests/fixtures/go_repo/internal/types/auth_dto.go
// Package types contains request and response DTOs serialised by the
// handler layer, plus JWT claim types shared between handlers, middleware,
// and the auth service.
package types

import "github.com/golang-jwt/jwt/v5"

// LoginRequest is the body of POST /auth/login.
type LoginRequest struct {
	Email    string `json:"email" validate:"required,email"`
	Password string `json:"password" validate:"required,min=8"`
}

// RefreshRequest is the body of POST /auth/refresh.
type RefreshRequest struct {
	RefreshToken string `json:"refresh_token" validate:"required"`
}

// TokenPair is the JSON shape returned by /auth/login and /auth/refresh.
type TokenPair struct {
	AccessToken  string `json:"access_token"`
	RefreshToken string `json:"refresh_token"`
	ExpiresIn    int64  `json:"expires_in"`
	TokenType    string `json:"token_type"`
}

// TokenType discriminates between access and refresh JWTs.
type TokenType string

const (
	AccessToken  TokenType = "access"
	RefreshToken TokenType = "refresh"
)

// Claims is the custom JWT payload embedded in every issued token.
// `Subject` (in RegisteredClaims) holds the user ID.
type Claims struct {
	jwt.RegisteredClaims
	Type TokenType `json:"typ"`
}

// IsAccess reports whether these claims belong to an access token.
func (c *Claims) IsAccess() bool {
	return c != nil && c.Type == AccessToken
}

// IsRefresh reports whether these claims belong to a refresh token.
func (c *Claims) IsRefresh() bool {
	return c != nil && c.Type == RefreshToken
}

File: tests/fixtures/rust_repo/src/dto/auth_dto.rs
//! Request / response types for `/auth/login` and `/auth/refresh`.

use serde::{Deserialize, Serialize};
use uuid::Uuid;
use validator::Validate;

/// JSON body for `POST /auth/login`.
#[derive(Debug, Clone, Deserialize, Validate)]
pub struct LoginRequest {
    #[validate(email)]
    pub email: String,
    #[validate(length(min = 8))]
    pub password: String,
}

/// JSON body for `POST /auth/refresh`.
#[derive(Debug, Clone, Deserialize)]
pub struct RefreshRequest {
    pub refresh_token: String,
}

/// JSON response envelope returned by both `/auth/login` and `/auth/refresh`.
///
/// `expires_in` is reported in seconds and tracks `access_token` (not the
/// refresh token, which lives much longer).
#[derive(Debug, Clone, Serialize)]
pub struct TokenPair {
    pub access_token: String,
    pub refresh_token: String,
    pub expires_in: u64,
    pub token_type: &'static str,
}

/// Discriminator for what kind of JWT we're minting. Embedded as the `typ`
/// claim so a refresh token can't accidentally be presented as an access
/// token (or vice versa).
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum TokenType {
    Access,
    Refresh,
}

/// Claims payload signed inside every JWT we issue.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Claims {
    pub sub: Uuid,
    pub iat: i64,
    pub exp: i64,
    pub typ: TokenType,
}
